src/sagemaker/session.py

```diff
@@ -170,6 +170,14 @@ def default_bucket(self):
             elif error_code == 'OperationAborted' and 'conflicting conditional operation' in message:
                 # If this bucket is already being concurrently created, we don't need to create it again.
                 pass
+            elif error_code == 'TooManyBuckets':
+                try:
+                    s3.meta.client.head_bucket(Bucket=default_bucket)
```

Comment:
This is a little esoteric - if we remove the logging below, can you add some developer documentation:

```# Succeed if the default bucket exists```